stacks-signer/src/signer.rs

```diff
@@ -415,24 +423,34 @@ impl Signer {
         stacks_client: &StacksClient,
         current_reward_cycle: u64,
     ) {
-        let coordinator_id = self.get_coordinator(current_reward_cycle).0;
         match &self.state {
             State::Idle => {
+                let Some(command) = self.commands.pop_front() else {
```

Comment:
Not required, but one way this could be made a bit more robust IMHO is to avoid doing the `pop_front()` until you've actually carried out the command.  You're just peeking the command for most of this `State::Idle` case; you can defer popping the command until the call to `execute_command` at the end.  Then, if this code body gets bigger, the future authors don't have to worry about "putting back" the command that got popped at the start of this block.  Just a thought :)

Reply:
I purposefully did it this way because we cannot borrow commands while simultaneously borrowing self in the execute command call.